sql/core/src/main/scala/org/apache/spark/sql/execution/analysis/DetectAmbiguousSelfJoin.scala

```diff
@@ -137,6 +137,9 @@ class DetectAmbiguousSelfJoin(conf: SQLConf) extends Rule[LogicalPlan] {
           }
           condition.toSeq.flatMap(getAmbiguousAttrs)
 
+        case _ if plan.find(_.isInstanceOf[Join]).isEmpty =>
+          Nil  // If there's no join, there's no self-join.
```

Comment:
can we do it at the beginning to make it clear?
```
if (!conf.getConf(SQLConf.FAIL_AMBIGUOUS_SELF_JOIN_ENABLED)) return plan
// ...
if (plan.find(_.isInstanceOf[Join]).isEmpty) return plan
```

Reply:
Yes, we can. I wanted to make a minimised fix with running the metadata cleanup logic but maybe I was worried too much.